features/accommodations/creation/PricingForm.tsx

```diff
@@ -1,14 +1,113 @@
-import { type FC } from "react";
+import { useState, type FC } from "react";
 import { type CreateAccommodation } from "../createAccommodation";
+import { type PriceDiff } from "../AccommodationModels";
+import { Button } from "../../../core/components/Button";
 import RadioGroup from "../../../core/components/RadioGroup";
+import { MONTHS, WEEKDAYS, parseCronExpression, toCronExpression } from "../../../core/utils/cron";
+import Select from "react-select";
+import { TrashIcon } from "@heroicons/react/24/outline";
+
+const monthOptions = MONTHS.map((m, i) => ({ value: i + 1, label: m }));
+const monthDayOptions = [...Array(31).keys()].map((i) => ({ value: i + 1, label: i + 1 }));
+const weekDayOptions = WEEKDAYS.map((d, i) => ({ value: i + 1, label: d }));
 
 type PricingFormProps = {
   updateFields: (fields: Partial<CreateAccommodation>) => void;
   price: number;
   isPerGuest: boolean;
+  priceDiffs: PriceDiff[];
 };
 
-export const PricingForm: FC<PricingFormProps> = ({ updateFields, price, isPerGuest }) => {
+const PriceDiffForm: FC<{ priceDiff: PriceDiff; onChange: () => void; onDelete: () => void }> = ({
+  priceDiff,
+  onChange,
+  onDelete,
+}) => {
+  const [when, setWhen] = useState(parseCronExpression(priceDiff.when || "0 0 * * *"));
+
+  const updateWhen = (field: "monthDays" | "months" | "weekDays", newValues: number[]) => {
+    let w = { ...when, [field]: newValues };
+    priceDiff.when = toCronExpression(w.monthDays, w.months, w.weekDays);
```

Comment:
You shouldn't usually change prop directly, can you pass it through onChange and then update?